### ISSUE TITLE
flask-admin is not compatible with jinja2 3.1.0 and later

### ISSUE THREAD

**Issue by blaise-io** [pallets-eco/flask-admin#2245]
```
File "/Users/blaise/app/.env/lib/python3.8/site-packages/flask_admin/helpers.py", line 2, in <module>
from jinja2 import contextfunction
ImportError: cannot import name 'contextfunction' from 'jinja2' (/Users/blaise/app/.env/lib/python3.8/site-packages/jinja2/__init__.py)
```

[Jinja2 removed `contextfunction` in 3.1.0](https://github.com/pallets/jinja/compare/3.0.3..3.1.0#diff-f41ee9a4b3adc33ac05af026d43d4a5958da45063e8e8bddaa923e759eaecc54L34)

**Comment by fabianhoward** [comment]
Have you tried reproducing this issue using v1.6.0 of this library? https://github.com/flask-admin/flask-admin/pull/2110 looks like it probably fixed your issue, which is included in v1.6.0.

**Comment by akashgreninja** [comment]
Hey bud you might wanna read this https://flask-login.readthedocs.io/en/latest/
